v3/src/components/codap-modal.tsx

```diff
@@ -0,0 +1,63 @@
+import { Modal,
+  ModalOverlay, ModalContent, ModalHeader, ModalCloseButton, ModalBody, ModalFooter,
+  Button } from "@chakra-ui/react"
+import React from "react"
+
+import "./codap-modal.scss"
+
+interface IModalButton {
+  className?: string;
+  label: string | React.FC<any>;
+  isDefault?: boolean;
+  isDisabled?: boolean;
+  onClick?: (() => void) | (() => boolean); // close dialog on falsy return value
+}
+
+interface IProps<IContentProps> {
+  children?: any
+  isOpen: boolean
+  onClose: () => void
+  className?: string
+  title: string
+  Icon?: React.FC<any>
+  Content: React.FC<IContentProps>
```

Comment:
For my benefit: How can IContentProps be part of the definition of IContentProps?

Reply:
IContentProps is a template type that allows Content to specify what type it will be. https://www.typescriptlang.org/docs/handbook/2/objects.html#generic-object-types

Reply:
IContentProps is a template type that allows Content to specify what type it will be. [https://www.typescriptlang.org/docs/handbook/2/objects.html#generic-object-types](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html)